lib/internal/modules/esm/load.js

```diff
@@ -141,22 +141,34 @@ function defaultLoadSync(url, context = kEmptyObject) {
 
   throwIfUnsupportedURLScheme(urlInstance, false);
 
+  let shouldBeReloadedByCJSLoader = false;
   if (urlInstance.protocol === 'node:') {
     source = null;
-  } else if (source == null) {
-    ({ responseURL, source } = getSourceSync(urlInstance, context));
-    context.source = source;
-  }
+    format ??= 'builtin';
+  } else if (format === 'addon') {
+    // Skip loading addon file content. It must be loaded with dlopen from file system.
+    source = null;
+  } else {
+    if (source == null) {
+      ({ responseURL, source } = getSourceSync(urlInstance, context));
+      context = { __proto__: context, source };
```

Comment:
This looks like a mistake: context is recursively itself + `source`? Should it be this instead?

```suggestion
      context = { __proto__: null, ...context, source };
```

Reply:
This is intentional. The effect would be similar, just instead of copying it uses the prototype chain to inherit the same properties.

Reply:
There are a couple of usage of this, most significantly this is what `defaultLoad` has been doing for years:

https://github.com/nodejs/node/blob/ea586599a500f5c20c058e0022ca7bbb8d93e9c9/lib/internal/modules/esm/load.js#L89

The benefit is that there'll be less copying (so slight perf boost, and lower memory usage. It can add up for a hot path like this).